pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -143,6 +143,9 @@ message MessageMetadata {
 	optional int32 num_chunks_from_msg = 27;
 	optional int32 total_chunk_msg_size = 28;
 	optional int32 chunk_id = 29;
+
+    // Indicate if the message key is set
+    optional bool null_key = 30 [default = false];
```

Comment:
It's better to use null_partition_key here since we have partition_key and ordering key in the proto. Use null_partition_key will be more accurate.